annotation/tests/test_tasks_services.py

```diff
@@ -295,3 +338,983 @@ def test_validate_user_actions_invalid_states(
+
+
+def test_finish_validation_task(mock_session: MagicMock):
+    mock_task = create_task(TaskStatusEnumSchema.ready)
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.with_for_update.return_value = mock_query
+    mock_query.update.return_value = None
+
+    finish_validation_task(mock_session, mock_task)
+
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+
+    mock_query.with_for_update.assert_called_once()
+    mock_query.update.assert_called_once_with(
+        {ManualAnnotationTask.status: TaskStatusEnumSchema.finished},
+        synchronize_session="fetch",
+    )
+    mock_session.commit.assert_called_once()
+
+
+def test_count_annotation_tasks(mock_session: Mock):
+    mock_task = create_task(TaskStatusEnumSchema.ready)
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.count.return_value = 5
+
+    result = count_annotation_tasks(mock_session, mock_task)
+
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+    mock_query.count.assert_called_once()
+
+    assert result == 5
+
+
+@pytest.fixture
+def mock_task_revisions() -> List[AnnotatedDoc]:
+    mock_revision = MagicMock(spec=AnnotatedDoc)
+    mock_revision.pages = {"1": ["data1"], "2": ["data2"]}
+    mock_revision.failed_validation_pages = [1, 2]
+    mock_revision.validated = [1, 2]
+    return [mock_revision]
+
+
+def test_get_task_revisions(
+    mock_session: Mock, mock_task_revisions: List[AnnotatedDoc]
+):
+    tenant = "test_tenant"
+    job_id = 1
+    task_id = 1
+    file_id = 1
+    task_pages = [1]
+
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.order_by.return_value = mock_query
+    mock_query.all.return_value = mock_task_revisions
+
+    result = get_task_revisions(
+        mock_session, tenant, job_id, task_id, file_id, task_pages
+    )
+
+    mock_session.query.assert_called_once_with(AnnotatedDoc)
+    mock_query.all.assert_called_once()
+
+    assert len(result) == 1
+    assert result[0].pages == {"1": ["data1"]}
+    assert result[0].failed_validation_pages == [1]
+    assert result[0].validated == [1]
+
+
+def test_get_task_info(mock_session: Mock):
+    mock_task = create_task(TaskStatusEnumSchema.ready)
+    task_id = 1
+    tenant = "test_tenant"
+
+    mock_query = MagicMock()
+    mock_session.query.return_value = mock_query
+    mock_query.filter.return_value = mock_query
+    mock_query.first.return_value = mock_task
+
+    result = get_task_info(mock_session, task_id, tenant)
+
+    mock_session.query.assert_called_once_with(ManualAnnotationTask)
+    mock_query.first.assert_called_once()
+    assert result == mock_task
+
+
+@pytest.fixture
+def annotated_file_pages() -> List[int]:
+    return [1, 2, 3]
```

Comment:
Too simple and unnecessary, since it is being used in only one test case, in which it could be passed in to the function being tested as a named value.